### PR TITLE
Filename variables

### PR DESCRIPTION
**Pull request by pontus-osterdahl**
This pull request adds some of the functionality requested in [#4916 ](https://github.com/kitodo/kitodo-production/issues/4916) by introducing two placeholders for use in METS-paths in folder configurations. 

- "filename": the referenced filename (without path).
- "basename": the basename of the file (without path). 

The METS-path "http://www.example.com/content/(basename)/default/(filename)" should now in the exported METS be resolved as "http://www.example.com/content/Image00001/default/Image00001.jpg".

The METS-path specifies the URL by which an image can be retrieved by the DFG-Viewer. Currently, the filename is added to the end of the URL, but information such as the filename may be necessary at other positions when external image servers are used. The addition of placeholders will hopefully reduce the need for post-processing.

Notes:
   - The filename-placeholders are only used during export and will not be accessible e.g. in script-paths.
   - If a file-placeholder is replaced, the filename is not by default added to the end of the METS-path. The reason for this is to improve flexibility. The filename must in such cases, as in the example above, be added via a placeholder.
   - If the METS-path does not contain "(filename)" or "(basename)", the filename is by default added to the end of the METS-path.

### CODE REVIEW

File: Kitodo/src/main/java/org/kitodo/production/helper/VariableReplacer.java [hunk 1]
```diff
@@ -368,4 +396,121 @@ private String determineReplacementForFirstchild(Matcher variableFinder, String
...
+     *         invalid, the string found is returned and no replacement takes
+     *         place. If the variable cannot be replaced because the requested
+     *         data cannot be accessed, the empty string is returned.
+     */
+    private String determineReplacementForFilePlaceholder(Matcher variableFinder, String filename) {
+        switch (variableFinder.group(6)) {
+    	    case "filename":
+                return determineReplacementForFilename(variableFinder, filename);
+    	    case "filename_without_extension":
+                return determineReplacementForFilenameWithoutExtension(variableFinder, filename);
```
Comment (matthias-ronge) [inline Kitodo/src/main/java/org/kitodo/production/helper/VariableReplacer.java:468]: You can simplify this a lot (and get rid of two more functions) by using `org.apache.commons.io.FilenameUtils`:
```java
switch (variableFinder.group(6)) {
    case "filename":
        return variableFinder.group(1) + FilenameUtils.getName(filename);
    case "basename":
        return variableFinder.group(1) + FilenameUtils.getBaseName(filename);
    default:
        ...
```

**Review by matthias-ronge — CHANGES_REQUESTED** [review]
In summary, a good and previously missing functionality! But the code can be improved in some places.

- `(filename_without_extension)` is long. I suggest `(basename)` instead.
- Make sure your `.java` files do not contain tabs, only spaces. (Configure your editor or code formatter settings.)
- Whenever possible, please do not duplicate (copy-paste) code. You can refactor the code if necessary (moving existing code into another method, etc.).
- Javadoc: The method description (the text before the first `@param` or the `@return` label) needs to contain a period (`.`). This is because everything before the first period is copied into the summary table on top of the Javadoc page. This sentence should be a verbal phrase (a phrase starting with a verb, mentally completing the sentence “This function …”) Good: “[This function] Replaces filename variables with the file name.” Not: “[This function] If a filename is to be determined, it is determined here.” The `@param` and `@return` descriptions do not need a period at the end. They never go into the summary table. BTW, you don’t need to write Javadoc for `private` functions as they do not show in the Javadoc at all. If there is something really important to explain about a `private` function, you can point this out using a non-Javadoc comment (starting the comment with `/*` instead `/**`).

See also my detail comments below.

**Comment by pontus-osterdahl** [comment]
Thank you @matthias-ronge for your thorough review and helpful suggestions! They should be included now.

**Comment by matthias-ronge** [comment]
When you think you've incorporated all changes, you can click this button …

the button

… to request a new review. The pull request then reappears for me in [the list of assigned pull request (you can bookmark this link for own list)](https://github.com/kitodo/kitodo-production/pulls/review-requested/@me).

**Review by matthias-ronge — APPROVED** [review]
That looks good! (I haven't downloaded and tested it, I assume it works.)

**Pull request merged**